src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2201 @@
+		}
+		return destinationLabel;
+	}
+
+	private JComboBox<MovementType> getJComboCharge() {
+		MovementType movementSelected = null;
+		if (chargeCombo == null) {
+			chargeCombo = new JComboBox<>();
+			try {
+				List<MovementType> movementTypes = movTypeManager.getMedicalDsrStockMovementType();
+				chargeCombo.addItem(null);
+				for (MovementType movType : movementTypes) {
+					if (movType.getType().equals("+")) {
+						chargeCombo.addItem(movType);
+						if (inventory != null && movType.getCode().equals(inventory.getChargeType())) {
+							movementSelected = movType;
+						}
+					}
+				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (inventory != null) {
+				chargeCombo.setSelectedItem(movementSelected);
+				chargeType = movementSelected;
+			}
+			chargeCombo.addActionListener(actionEvent -> {
+				chargeType = (MovementType) chargeCombo.getSelectedItem();
+			});
+		}
+		return chargeCombo;
+	}
+
+	private JComboBox<MovementType> getJComboDischarge() {
+		MovementType movementSelected = null;
+		if (dischargeCombo == null) {
+			dischargeCombo = new JComboBox<>();
+			try {
+				List<MovementType> movementTypes = movTypeManager.getMedicalDsrStockMovementType();
+				dischargeCombo.addItem(null);
+				for (MovementType movType : movementTypes) {
+					if (movType.getType().equals("-")) {
+						dischargeCombo.addItem(movType);
+						if (inventory != null && movType.getCode().equals(inventory.getDischargeType())) {
+							movementSelected = movType;
+						}
+					}
+				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (inventory != null) {
+				dischargeCombo.setSelectedItem(movementSelected);
+				dischargeType = movementSelected;
+			}
+			dischargeCombo.addActionListener(actionEvent -> {
+				dischargeType = (MovementType) dischargeCombo.getSelectedItem();
+			});
+		}
+		return dischargeCombo;
+	}
+
+	private JComboBox<Supplier> getJComboSupplier() {
+		Supplier supplierSelected = null;
+		if (supplierCombo == null) {
+			supplierCombo = new JComboBox<>();
+			try {
+				List<Supplier> suppliers = supplierManager.getList();
+				supplierCombo.addItem(null);
+				for (Supplier supplier : suppliers) {
+					supplierCombo.addItem(supplier);
+					if (inventory != null && supplier.getSupId() == inventory.getSupplier()) {
+						supplierSelected = supplier;
+					}
+				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (inventory != null) {
+				supplierCombo.setSelectedItem(supplierSelected);
+				supplier = supplierSelected;
+			}
+			supplierCombo.addActionListener(actionEvent -> {
+				supplier = (Supplier) supplierCombo.getSelectedItem();
+			});
+		}
+		return supplierCombo;
+	}
+
+	private JComboBox<Ward> getJComboDestination() {
+		Ward destinationSelected = null;
+		if (destinationCombo == null) {
+			destinationCombo = new JComboBox<Ward>();
```

Comment:
```suggestion
			destinationCombo = new JComboBox<>();
```
replace type specification with `<>`